src/Resources/contao/classes/Backend.php

```diff
@@ -411,10 +414,40 @@ protected function getBackendModule($module)
 		}
 
 		// Custom action (if key is not defined in config.php the default action will be called)
-		elseif (\Input::get('key') && isset($arrModule[\Input::get('key')]))
+		elseif (\Input::get('key'))
 		{
-			$objCallback = \System::importStatic($arrModule[\Input::get('key')][0]);
-			$this->Template->main .= $objCallback->{$arrModule[\Input::get('key')][1]}($dc);
+			// Use the regular callback (BC)
+			if (isset($arrModule[\Input::get('key')])) {
+				@trigger_error('Using backend module callbacks has been deprecated and will no longer work in Contao 5.0. Use controllers to generate custom module actions.', E_USER_DEPRECATED);
+
+				$className  = $arrModule[\Input::get('key')][0];
+				$methodName = $arrModule[\Input::get('key')][1];
+				$objCallback = System::importStatic($className);
+				$this->Template->main .= $objCallback->$methodName($dc);
+			} elseif (isset($arrModule['controllers'][\Input::get('key')])) {
+				// Use a controller
+				$container = static::getContainer();
+
+				/** @var Request $request */
+				$request    = $container->get('request_stack')->getCurrentRequest();
+				$controller = $arrModule['controllers'][\Input::get('key')][0];
+				$strategy   = ($arrModule['controllers'][\Input::get('key')][1]) ?: 'inline';
```

Comment:
Most likely not, that's why `inline` is default. But why would we want to limit it?